modules/user_accounts/php/user_accounts.class.inc

```diff
@@ -40,102 +39,56 @@ class User_Accounts extends \NDB_Menu_Filter
     }
 
     /**
-     * Setup Variables
+     * Tells the base class that this page's provisioner can support the
+     * UserSiteMatch filter.
      *
-     * @note   Setup Variables
-     * @return void
+     * @return bool true
      */
-    function _setupVariables()
+    public function useSiteFilter() : bool
     {
-        $user =& \User::singleton();
-        // the base query
-        $query = " FROM users
-            LEFT JOIN user_psc_rel ON (user_psc_rel.UserID=users.ID)
-            LEFT JOIN psc ON (user_psc_rel.CenterID=psc.CenterID)
-            WHERE 1=1";
-        if (!$user->hasPermission('user_accounts_multisite')) {
-            $site_arr = implode(",", $user->getCenterIDs());
-            $query   .= " AND FIND_IN_SET(user_psc_rel.CenterID,'$site_arr')";
-        }
-
-        // set the class variables
-        $this->columns      = array(
-            "COALESCE(GROUP_CONCAT(
-                                psc.Name SEPARATOR '; '),'Not Assigned'
-                                ) AS Sites",
-            'users.UserID AS Username',
-            'Real_name AS Full_name',
-            'Email',
-            'Active',
-            'Pending_approval',
-        );
-        $this->query        = $query;
-        $this->order_by     = 'Username';
-        $this->group_by     = 'users.ID';
-        $this->validFilters = array(
-            'user_psc_rel.CenterID',
-            'users.UserID',
-            'users.Real_name',
-            'users.Email',
-            'users.Active',
-            'users.Examiner',
-            'users.Pending_approval',
-        );
-
-        $this->formToFilter = array(
-            'centerID'  => 'user_psc_rel.CenterID',
-            'active'    => 'users.Active',
-            'userID'    => 'users.UserID',
-            'real_name' => 'users.Real_name',
-            'email'     => 'users.Email',
-            'pending'   => 'users.Pending_approval',
-        );
+        return true;
     }
 
     /**
-     * Does the setup required for this page. By default, sets up elements
-     * that are common to every type of page. May be overridden by a specific
-     * page or specific page type.
+     * Tells the base class that this page's provisioner can support the
+     * UserProjectMatch filter.
      *
-     * @return void
+     * @return bool true
      */
-    function setup()
+    public function useProjectFilter() : bool
     {
-        parent::setup();
-
-        // create user object
-        $user =& \User::singleton();
+        return false;
```

Comment:
```suggestion
        return true;
```

Users have projects, this should have project filters.

Reply:
okay

Reply:
are you going to merge this change?

Reply:
Yeah! It's just from my understanding this needs to involve other changes as well like including a getProjects() function, and querying the projects as well. I was going to get around to doing that soon.